src/renderers/dom/ReactDOM.js

```diff
@@ -133,6 +133,13 @@ if (__DEV__) {
       }
     }
   }
+
+  // Check if using Polymer and we are not using native shadow
+  var shadyDom = window.Polymer !== undefined && !window.Polymer.useNativeShadow;
```

Comment:
We would need to check that `window` exists here, as it may not on the server, for example. I think usually `ExecutionEnvironment.canUseDOM` is the check we use (you can search for it in the source).